cli/src/config.rs

```diff
@@ -122,8 +123,23 @@ pub fn read_configuration(path: &Path) -> Result<Config> {
     };
 
     // If an api token has been set in the environment, prefer that
-    if let Ok(key) = env::var("PHYLUM_API_KEY") {
-        config.auth_info.offline_access = Some(RefreshToken::new(key));
+    match env::var("PHYLUM_API_KEY") {
+        Ok(key) if !key.is_empty() => {
+            config.auth_info.offline_access = Some(RefreshToken::new(key));
+        },
+        Ok(_) => {
+            print_user_warning!("Ignoring empty PHYLUM_API_KEY");
+        },
+        Err(VarError::NotUnicode(_)) => {
+            print_user_warning!("Ignoring invalid PHYLUM_API_KEY");
+        },
+        Err(VarError::NotPresent) => {},
+    }
+
+    // The code that checks if we have a token expects `Some(token)` to always be a
+    // usable token.
+    if config.auth_info.offline_access.as_ref().map(|t| t.as_str().is_empty()).unwrap_or_default() {
```

Comment:
There are many shorter ways to write this condition. Here is one possibility.

```suggestion
    if config.auth_info.offline_access.as_ref().map(|t| t.as_str()) == Some("") {
```